src/cli/cmd/mod.rs

```diff
@@ -269,6 +288,50 @@ pub(crate) fn print_json<T: Serialize>(value: T) -> Result<(), FhError> {
     Ok(())
 }
 
+struct FlakeOutputRef {
+    org: String,
+    flake: String,
+    version: String,
+    output: String,
+}
+
+impl TryFrom<String> for FlakeOutputRef {
+    type Error = FhError;
+
+    fn try_from(flake_ref: String) -> Result<Self, Self::Error> {
+        let parts: Vec<&str> = flake_ref.split('#').collect();
```

Comment:
```suggestion
        let parts: Vec<&str> = flake_ref.splitn(2, '#').collect();
```

and perhaps splitn(3, '/') below?

Reply:
I'm open to it but I'm not sure it really adds much in this case, as the arguments would still need to be checked one by one